contracts/base/OnlySelfCallable.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.18;
+
+abstract contract OnlySelfCallable {
+    error InvalidSender(address sender);
+    error InvalidCalldataLength();
+
+    modifier onlySelf() {
+        checkCallerisSender();
+        _;
+    }
+
+    function checkCallerisSender() private view {
```

Comment:
This method is severely lacking comments. We use assembly and do length checks, both should be well commented and if possible linked to standards this is based on.